### DISCUSSION TITLE
How to parse an rss feed

### DISCUSSION THREAD

**Discussion by pglevy** [reflex-dev/reflex#3674] [Q&A]
I'm having trouble with data types in State (I think).

I'm trying to do something like this with an rss feed (from regular python):
```
# main.py
import feedparser

feed = "https://feedparser.readthedocs.io/en/latest/examples/rss20.xml"

parsedFeed = feedparser.parse(feed)

for i in range(10):
    entry = parsedFeed.entries[i]
    print(entry.title)
    print(entry.description)
    # etc
```

My two main questions when trying to do this in Reflex:
1. How do I set a state variable to the value of the parsed feed? (I don't see many examples beyond a few primitive types.)
2. Do I do the iteration in the State function? Or use rx.foreach with the components?

```
# state.py
class State(rx.State):

    feed: str = "https://feeds.transistor.fm/acquired"

    parsedFeed: ??? = feedparser.parse(feed)
```

Thanks for any help!

**Comment by abulvenz** [comment]
You can make it a computed var.
```python
class FeedItem(rx.Base):
  url: str
  text: str
  ...

class State(rx.State):
  feed_url=...
  @rx.var
  def parsedFeed(self) -> list[FeedItem]:
     result = []
     for article in feedparser.parse(self.feed):
       result.append(FeedItem(...))
     return result
```
Have a look at the `cache` flag and invalidation timeouts.
The important thing is that you end up with something that reflex knows how to serialize for the frontend.
So you can either create serializers using `@serializer` or use something derived from `rx.Base`.

**Comment by pglevy** [comment]
OK, thanks! I'll give that a shot.
